packages/steppers/src/elements/components/StepLabel.tsx

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { forwardRef, HTMLAttributes } from 'react';
+import PropTypes from 'prop-types';
+import CheckCircleStrokeIcon from '@zendeskgarden/svg-icons/src/16/check-sm-stroke.svg';
+import { StyledLabel, StyledIcon, StyledIconFlexContainer } from '../../styled';
+import { useStepContext, useStepperContext } from '../../utils';
+
+interface IStepLabel {
+  icon?: React.ReactNode;
+}
+
+export const StepLabel = forwardRef<HTMLDivElement, IStepLabel & HTMLAttributes<HTMLDivElement>>(
+  (props, ref) => {
+    const { currentStepIndex } = useStepContext();
+    const { activeIndex, isHorizontal } = useStepperContext();
+    const numericStep = currentStepIndex + 1;
+    const stepIcon = props.icon || numericStep;
+    const isActive = activeIndex === currentStepIndex;
+    const isCompleted = activeIndex > currentStepIndex;
+    const styledIcon = (
+      <StyledIcon isActive={isActive} isHorizontal={isHorizontal}>
+        {isCompleted ? <CheckCircleStrokeIcon /> : stepIcon}
+      </StyledIcon>
+    );
+
+    return (
+      <StyledLabel ref={ref} isActive={isActive} isHorizontal={isHorizontal} {...props}>
+        {isHorizontal ? (
+          <StyledIconFlexContainer>{styledIcon}</StyledIconFlexContainer>
```

Comment:
Only applying flex to the `StyledLabel` worked across all browsers except IE11 — in IE11, this caused the long labels to render onto a single line. To address this issue for IE but also have the layout correct for other browsers, I restructured the HTML/CSS so that Step Icons are centered within their flex container in horizontal steppers.